src/jitlayers.h

```diff
@@ -42,7 +42,14 @@
 // and feature support (e.g. Windows, JITEventListeners for various profilers,
 // etc.). Thus, we currently only use JITLink where absolutely required, that is,
 // for Mac/aarch64.
-#if defined(_OS_DARWIN_) && defined(_CPU_AARCH64_) || defined(_COMPILER_ASAN_ENABLED_) || defined(JL_FORCE_JITLINK)
+// #define JL_FORCE_JITLINK
+
+#if defined(_COMPILER_ASAN_ENABLED_) || defined(_COMPILER_MSAN_ENABLED_) || defined(_COMPILER_TSAN_ENABLED_)
+# define HAS_SANITIZER
+#endif
+// The sanitizers don't play well with our memory manager
+
+#if defined(_OS_DARWIN_) && defined(_CPU_AARCH64_) || defined(JL_FORCE_JITLINK) || defined(HAS_SANITIZER)
```

Comment:
```suggestion
#if defined(_OS_DARWIN_) && defined(_CPU_AARCH64_) || defined(JL_FORCE_JITLINK)
```

Right? But I think we can remove that entire check soon.

Reply:
Actually I think it needs to be `|| JL_LLVM_VERSION >= 150000 && defined(HAS_SANITIZER)`; we want JITLink if we're using the newpm sanitizer since we apparently can't prevent the sanitizers from recursing the memory manager, and we want the newpm sanitizer when we're on LLVM 15 or higher.